bf: ZENKO-1183 fix scheduled crr resume

philipyoo opened the pull request:
Changes in this PR:
- Fix wrong property name used on init 😨 
- Re-schedule resume when a triggered scheduled resume
  fails. Only error that can occur is if update to
  zookeeper fails

philipyoo committed:
bf: ZENKO-1183 fix scheduled crr resume

Changes in this commit:
- Fix wrong property name used on init
- Re-schedule resume when a triggered scheduled resume
  fails. Only error that can occur is if update to
  zookeeper fails

bert-e commented:
# Hello philipyoo,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @vrancurel

* @rahulreddy

* @bennettbuchanan

* @jonathan-gramain

bennettbuchanan commented on extensions/replication/queueProcessor/QueueProcessor.js:
```diff
@@ -383,26 +383,35 @@ class QueueProcessor extends EventEmitter {
+        function triggerResume() {
+            this._updateZkStateNode('scheduledResume', null, err => {
+                if (err) {
+                    this.logger.error('error occurred saving state ' +
```
Maybe add a log indicating it is being rescheduled for a minute later.

philipyoo replied:
I'll update the message to:
`error occurred saving state to zookeeper for resuming a scheduled resume. Retry again in 1 minute`

bennettbuchanan commented on extensions/replication/queueProcessor/QueueProcessor.js:
```diff
@@ -383,26 +383,35 @@ class QueueProcessor extends EventEmitter {
+                if (err) {
+                    this.logger.error('error occurred saving state ' +
+                    'to zookeeper', {
+                        method: 'QueueProcessor.scheduleResume',
```
Let's add the error here as well?

rahulreddy commented on extensions/replication/queueProcessor/QueueProcessor.js:
```diff
@@ -383,26 +383,37 @@ class QueueProcessor extends EventEmitter {
 
     scheduleResume(date) {
+        function triggerResume() {
+            this._updateZkStateNode('scheduledResume', null, err => {
```
This node name `scheduledResume` is the same one which was mis-spelt, correct? Why not use a constant 
`const RESUME_NODE = 'scheduledResume';`

philipyoo committed:
improvement: ZENKO-1183 use constants state props

Changes in this commit:
- move zookeeper crr state path constant to the replication
  constants file
- use const SCHEDULE_PROP in init file for schedule resume
  property
- add extra validation in helper method:
  `QueueProcessor._updateZkStateNode` to validate zk state
  property to be a valid property name

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.0`



The following branches will **NOT** be impacted:

* `development/7.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of Release Engineering will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of Release Engineering now.

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.0`



The following branches have **NOT** changed:

* `development/7.4`



Please check the status of the associated issue ZENKO-1183.

Goodbye philipyoo.
